index.bs

```diff
@@ -1705,46 +1705,56 @@ When this operation is invoked, the authenticator must perform the following pro
-    this [=public key credential|credential=]. The prompt for obtaining this [=user consent|consent=] may be shown by the
+1. Prompt the user to select a [=public key credential|credential=] |selectedCredential| from |allowCredentialDescriptorList|.
+    Obtain [=user consent=] for using |selectedCredential|. The prompt for obtaining this [=user consent|consent=] may be shown
+    by the
     [=authenticator=] if it has its own output capability, or by the user agent otherwise.
-1. Process all the supported extensions requested by the client, and generate the [=authenticator data=] as specified in
-    [[#sec-authenticator-data]], though without [=attestation data=]. Concatenate this [=authenticator data=] with the [=hash of
-    the serialized client data=] to generate an [=assertion signature=] using the [=credential private key|private key=] of the
-    selected [=public key credential|credential=] as shown in [Figure 2](#fig-signature), below. A simple, undelimited
+1. Let |processedExtensions| be the result of [=authenticator extension processing=] for each supported extension
+    identifier/input pair in |extensions|.
+1. Let |authenticatorData| be the byte array specified in [[#sec-authenticator-data]] without [=attestation data=] but with any
+    |processedExtensions|.
+1. Let |signature| be the [=assertion signature=] of the concatenation <code>|authenticatorData| || |hash|</code> using the
+    [=credential private key|private key=] of |selectedCredential| as shown in [Figure 2](#fig-signature), below. A simple,
+    undelimited
     concatenation is safe to use here because the [=authenticator data=] describes its own length. The [=hash of the serialized
     client data=] (which potentially has a variable length) is always the last element.
-1. If any error occurred while generating the [=assertion signature=], return an error code equivalent to "{{UnknownError}}" and
-    terminate the operation.
 
-<figure id="fig-signature">
-    <img src="images/fido-signature-formats-figure2.svg"/>
-    <figcaption>Generating an [=assertion signature=].</figcaption>
-</figure>
+    <figure id="fig-signature">
+        <img src="images/fido-signature-formats-figure2.svg"/>
+        <figcaption>Generating an [=assertion signature=].</figcaption>
+    </figure>
 
-On successful completion, the authenticator returns to the user agent:
-- The identifier of the credential (credential ID) used to generate the [=assertion signature=].
-- The [=authenticator data=] used to generate the [=assertion signature=].
-- The [=assertion signature=].
-- The user handle associated with the credential used to generate the [=assertion signature=].
+1. If any error occurred while generating the [=assertion signature=], return an error code equivalent to "{{UnknownError}}" and
+    terminate the operation.
+1. Return to the user agent:
+    - |selectedCredential|'s ID
```

Comment:
i think it is important here to s/ID/credential ID/  -- we are consistently (more or less) using that term and need to more formally define it, so being consistent will help with the latter.

Reply:
Done.